src/middlewared/middlewared/plugins/pool.py

```diff
@@ -4300,16 +4300,16 @@ async def __run(self, name, threshold):
         if pool['scan']['state'] == 'SCANNING':
             return False
 
-        history = (await run('zpool', 'history', name, encoding='utf-8')).stdout
-        for match in reversed(list(RE_HISTORY_ZPOOL_SCRUB.finditer(history))):
+        last_scrubs = (await run('sh', '-c', f'zpool history {shlex.quote(name)} | grep "zpool scrub"', encoding='utf-8')).stdout
+        for match in reversed(list(RE_HISTORY_ZPOOL_SCRUB.finditer(last_scrubs))):
             last_scrub = datetime.strptime(match.group(1), '%Y-%m-%d.%H:%M:%S')
             break
         else:
             # creation time of the pool if no scrub was done
-            for match in RE_HISTORY_ZPOOL_CREATE.finditer(history):
-                last_scrub = datetime.strptime(match.group(1), '%Y-%m-%d.%H:%M:%S')
-                break
-            else:
+            creation = (await run('zfs', 'get', '-Hpo', 'value', 'creation', name, encoding='utf-8')).stdout
```

Comment:
Why run additional command? Previously we had only one command running, now we have two. Using `egrep "zpool scrub|zpool create"` will include both scrub and creation lines.

Reply:
Yes, we have 2 commands now. I think the code is very clear now. The first part runs zpool history, filtering for `zpool scrub` and parsing it for last scrub time. And only if not found, the second part is run. The second command directly queries the pool (dataset creation, which is the same) creation. There is no need for regular expressions or any kind of parsing. And, that command is run once ever during one pool's lifetime, as next time the first part will win, so it is not really an overhead. And I think the code is simpler.